src/SignalRServiceExtension/TriggerBindings/Resolver/DefaultRequestResolver.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.Azure.SignalR.Serverless.Protocols;
+using Microsoft.Azure.WebJobs.Extensions.SignalRService.Exceptions;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal class DefaultRequestResolver : IRequestResolver
+    {
+        public bool ValidateContentType(HttpRequestMessage request)
+        {
+            var contentType = request.Content.Headers.ContentType.MediaType;
+            if (string.IsNullOrEmpty(contentType))
+            {
+                return false;
+            }
+            return contentType == Constants.JsonContentType || contentType == Constants.MessagePackContentType;
+        }
+
+        public bool ValidateSignature(HttpRequestMessage request, string accessToken)
+        {
+            //TODO: Add real signature validation
+            return true;
+        }
+
+        public bool TryGetInvocationContext(HttpRequestMessage request, out InvocationContext context)
+        {
+            if (!request.Headers.Contains(Constants.AsrsHubNameHeader) ||
+                !request.Headers.Contains(Constants.AsrsCategory) ||
+                !request.Headers.Contains(Constants.AsrsEvent) ||
+                !request.Headers.Contains(Constants.AsrsConnectionIdHeader))
+            {
+                context = null;
+                return false;
+            }
+
+            context = new InvocationContext();
+            context.ConnectionId = request.Headers.GetValues(Constants.AsrsConnectionIdHeader).First();
```

Comment:
use TryGetValue?

Reply:
If you think we should avoid multiple access to the same header value, I have a new apporach in the new commit. I think use GetValues().FirstOrDefault() is a much easier and cleaerer way than `TryGetValue`.